src/go/rpk/pkg/cli/cloud/login.go

```diff
@@ -49,39 +55,179 @@ Cloud client credentials can be used to login to Redpanda, they can be created
 in the Clients tab of the Users section in the Redpanda Cloud online interface. 
 client credentials can be provided in three ways, in order of preference:
 
-* In $HOME/.config/rpk/__cloud.yaml, in 'client_id' and 'client_secret' fields
+* In your rpk cloud auth, 'client_id' and 'client_secret' fields
 * Through RPK_CLOUD_CLIENT_ID and RPK_CLOUD_CLIENT_SECRET environment variables
 * Through the --client-id and --client-secret flags
 
 If none of these are provided, rpk will use the SSO method to login. 
 If you specify environment variables or flags, they will not be synced to the
-__cloud.yaml file unless the --save flag is passed. The cloud authorization 
+rpk.yaml file unless the --save flag is passed. The cloud authorization 
 token and client ID is always synced.
 `,
 		Run: func(cmd *cobra.Command, _ []string) {
-			cfg, err := params.Load(fs)
+			cfg, err := p.Load(fs)
 			out.MaybeDie(err, "unable to load config: %v", err)
+			y := cfg.VirtualRpkYaml()
+			auth := y.Auth(y.CurrentCloudAuth)
+			cc := auth.HasClientCredentials()
 
-			_, err = oauth.LoadFlow(cmd.Context(), fs, cfg, auth0.NewClient(cfg))
-			if e := (*oauth.BadClientTokenError)(nil); errors.As(err, &e) {
-				out.Die(`
-Unable to login into Redpanda Cloud: %v.
-
-You may need to clear your client ID and secret with 'rpk cloud logout --clear-credentials',
-and then re-specify the client credentials next time you log in.`, err)
+			_, err = oauth.LoadFlow(cmd.Context(), fs, cfg, auth0.NewClient(cfg.DevOverrides()))
+			if err != nil {
+				if e := (*oauth.BadClientTokenError)(nil); errors.As(err, &e) && cc {
+					fmt.Println(`You may need to clear your client ID and secret with 'rpk cloud logout --clear-credentials',
+and then re-specify the client credentials next time you log in.`)
+				}
+				os.Exit(1)
```

Comment:
We should print the error here, because if it's not an oauth.BadClientTokenError we won't print anything and exit(1)

out.Die(err) probably.

Reply:
Added as a line above this error, per your prior comment\